topk_test.go

```diff
@@ -443,3 +468,58 @@ func TestMarshalUnMarshal(t *testing.T) {
 	assert.EqualValues(t, sketch, tmp)
 
 }
+func TestCaseSensitive(t *testing.T) {
+	words := []string{
+		"Hello",
+		"hello",
+		"world",
+		"World",
+		"WORLD",
+		"world",
+		"goodbye",
+		"Goodbye",
+		"Goodbye",
+	}
+
+	sketch := New(1, true)
+
+	for _, w := range words {
+		sketch.Insert(w, 1)
+	}
+
+	keys := sketch.Keys() // should be either "world" or "Goodbye"
+	assert.Equal(t, 1, len(keys))
+	assert.Contains(t, []string{"world", "Goodbye"}, keys[0].Key)
+
+	sketch = New(1, false)
+	for _, w := range words {
+		sketch.Insert(w, 1)
+	}
+
+	keys = sketch.Keys() // should be either "world"
+	assert.Equal(t, 1, len(keys))
+	assert.Equal(t, "world", keys[0].Key)
+
+	buf := bytes.NewBuffer(nil)
+	err := sketch.Encode(buf)
+	assert.NoError(t, err)
+
+	decoded := New(1, false)
+	err = decoded.Decode(buf)
+	assert.NoError(t, err)
+	
+	if !reflect.DeepEqual(sketch, decoded) {
+		t.Error("they are not equal.")
+	}
+}
+
+func BenchmarkTopK(b *testing.B) {
+	words := loadWords()
+	for b.Loop() {
+		sketch := New(100, true)
+		for _, w := range words {
+			sketch.Insert(w, 1)
+		}
+		sketch.Keys()
+	}
```

Comment:
Should avoid including loadWords() allocs in report, and should ensure that the call to sketch.Keys() is not optimized out. Top 20 is probably more representative of a real-life workload than top 100.

```suggestion
	b.ResetTimer()
	
	dummy := 0
	for b.Loop() {
		sketch := New(20, true)
		for _, w := range words {
			sketch.Insert(w, 1)
		}
		dummy += len(sketch.Keys())
	}
	
	assert.True(b, dummy > 0)
```